cpp/examples/basic/CMakeLists.txt

```diff
@@ -1,16 +1,35 @@
-# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 
 cmake_minimum_required(VERSION 3.26.4)
 
+include(../set_cuda_architecture.cmake)
+
+# initialize cuda architecture
+rapids_cuda_init_architectures(basic_example)
+rapids_cuda_set_architectures(RAPIDS)
+
 project(
   basic_example
   VERSION 0.0.1
   LANGUAGES CXX CUDA
 )
 
 include(../fetch_dependencies.cmake)
-
 # Configure your project here
 add_executable(basic_example src/process_csv.cpp)
 target_link_libraries(basic_example PRIVATE cudf::cudf)
 target_compile_features(basic_example PRIVATE cxx_std_17)
+
+# Force set CMAKE_INSTALL_PREFIX from environment variable
+if(NOT $ENV{CMAKE_INSTALL_PREFIX} STREQUAL "")
```

Comment:
We should just require cmake 3.29 if you want to support CMAKE_INSTALL_PREFIX from the env, since that is the version where the variable is supported. 

Otherwise you are doing non standard behavior that isn't expected by CMake users.


Looking over the entire code, I expect that the following is going to simplify the logic and resolve your issues;

1. Yank all usages of `${CMAKE_INSTALL_PREFIX}` from the install commands. It is not needed and is an anti-pattern that is break `cmake --install --prefix`.
2. Yank all uages of trying to specify `CMAKE_INSTALL_PREFIX` from the env in the CMakeLists.txt
3. Yank all setting of the CMAKE_INSTALL_PREFIX in the conda recipe scripts.
4. Add something like a `--install` option to `cpp/examples/build.sh` which will call `cmake --install --prefix` using some logic where the install prefix location is deduced via something like: `INSTALL_PREFIX=${INSTALL_PREFIX:=${PREFIX:=${CONDA_PREFIX}}}`

Reply:
Thanks for the suggestions Robert. Incorporating these changes in the next commit.

Reply:
All done!